src/main/java/io/confluent/connect/hdfs/string/StringRecordWriterProvider.java

```diff
@@ -70,6 +70,12 @@ public void write(SinkRecord record) {
           writer.newLine();
         } catch (IOException e) {
           throw new ConnectException(e);
+        } finally {
+          try {
+            close();
+          } catch (Throwable t) {
+            log.error("Could not close Record Writer", t);
+          }
```

Comment:
This isn't correct here, because we're always calling `close()` after every call to `write(...)`. I was asking about the case when we catch the exception on 72, and possibly closing it here. But do we know whether we're always closing the record writer? If so, we don't need this here. (Please add a comment describing what you find.)

Reply:
You're right that this is incorrect, I got a little trigger-happy with the finally blocks. It appears that exceptions thrown from this method get caught in [TopicParititionWriter::write](https://github.com/confluentinc/kafka-connect-hdfs/blob/e13532024920e1433f741119427517583adf8e5e/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L409-L414).

The RecordWriter is closed in these cases:
* `HdfsSinkTask::start()` when an exception occurs while recovering the WAL
* `TopicParititonWriter::write()` when in state `SHOULD_ROTATE`
* `TopicPartitionWriter::write()` after a successful write
* `HdfsSinkTask::close()` when partitions are reassigned
* `HdfsSinkTask::stop()` when the task is stopped entirely

I definitely don't understand the retry strategy of the HDFS connector completely, but it appears that the error is being swallowed, so the framework does not trigger the sink record to be retried. If the write failed, all of the previous records that were in the process of being written are potentially lost as well, and need to be retried from the beginning. I can only assume that the connector already handles that properly, otherwise we would be seeing large data loss events whenever HDFS hiccups.

I think i'd err towards leaving the code as-is (as in rolling back my changes entirely), and not closing the writer or it's associated filesystem. They appear to be adequately cleaned up during task shutdown and partition reassignment, and I assume that the existing error handling hits those code-paths.